File: test-clients/java/template/src/test/java/theclient/EchoClientV2Test.java
package theclient;

import org.junit.jupiter.api.Test;
import theclient.v2.clients.echo.EchoClient;
import theclient.v2.models.Message;

import static org.junit.jupiter.api.Assertions.*;

public class EchoClientV2Test {
	private final String baseUrl = "http://localhost:8081";
	private final EchoClient client = new EchoClient(baseUrl);

	private final Message message = new Message(true, "the string");

	@Test
	public void echoBodyModel_responseIsEqualToRequest() {
		var response = client.echoBodyModel(message);
		assertEquals(message, response);
	}

	@Test
	public void echoBodyModel_doesntThrowException() {
		assertDoesNotThrow(() -> client.echoBodyModel(message));
	}
}

File: test-clients/kotlin/template/src/test/kotlin/theclient/EchoClientV2Test.kt
package theclient

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import theclient.v2.clients.echo.*
import theclient.v2.models.*

class EchoClientV2Test {
    private val client = EchoClient("http://localhost:8081")

    private val bodyMessage = Message(true, "the string")

    @Test
    fun echoBodyModel_responseIsEqualToRequest() {
        val response = client.echoBodyModel(bodyMessage)
        assertEquals(bodyMessage, response)
    }

    @Test
    fun echoBodyModel_doesntThrowException() {
        assertDoesNotThrow { client.echoBodyModel(bodyMessage) }
    }
}